### ISSUE TITLE
`col value outside range` with wildmenu

### ISSUE THREAD

**Issue by max397574** [hrsh7th/nvim-cmp#909]
### FAQ

- [X] I have checked the [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/15f08a8faa22d52480cdcb9ef9ca698120f04363/doc/cmp.txt#L616) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/hrsh7th/nvim-cmp/issues) and there are no open or closed issues with the same problem.

### Neovim Version

NVIM v0.8.0-dev+nightly-1651-ga8d5a9e81

### Minimal reproducible full config

not possible ??????

### Description

When I enter `lua print(vim.fn.get_c` I get this:
![Uploading Screenshot 2022-04-16 at 17.07.07.png…]()

### Steps to reproduce

-

### Expected behavior

no errors

### Actual behavior

errors

**Comment by max397574** [comment]
I'm trying to produce a minimal config without success so far

**Comment by Shougo** [comment]
Hi.

> When I enter lua print(vim.fn.get_c I get this:

The screenshot is not uploaded.

**Comment by max397574** [comment]
it no longer gives an error 😳
I tried it with a ton of different settings and it always was there.
And now it's just gone

**Comment by Shougo** [comment]
oh

**Comment by dansalvato** [comment]
This issue still exists, but I've fixed it. There are some cases with fuzzy matching where no match is found, but a match is still written to the matches table. That match has `word_match_start` value of -1, which can cause an invalid column number when used in `wildmenu_entries_view.lua`.

https://github.com/hrsh7th/nvim-cmp/blob/e7e2ef7031db8e0cfad7ad9916ec766ddd9dbb3b/lua/cmp/matcher.lua#L241

Change this line to:

```lua
  if matched and input_index > #input then
```

Below are screenshots from before and after I made this change, to help demonstrate the issue.


![2023-01-31_12-02-45](https://user-images.githubusercontent.com/7810794/215857338-60ad3552-01ea-4f74-b822-bf6986476f77.png)
![2023-01-31_12-02-17](https://user-images.githubusercontent.com/7810794/215857355-5e3d0a62-e0a9-4cb0-b346-57e46b9a95f5.png)

**Comment by Shougo** [comment]
You should create the PR.

**Comment by dansalvato** [comment]
This is fixed in #1420. By request of @hrsh7th, here is a simple test case that would always trigger the error.

`cmp-repro.vim`:
```vim
if has('vim_starting')
  set encoding=utf-8
endif
scriptencoding utf-8

if &compatible
  set nocompatible
endif

let s:plug_dir = expand('/tmp/plugged/vim-plug')
if !filereadable(s:plug_dir .. '/plug.vim')
  execute printf('!curl -fLo %s/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim', s:plug_dir)
end

execute 'set runtimepath+=' . s:plug_dir
call plug#begin(s:plug_dir)
Plug 'hrsh7th/nvim-cmp'
Plug 'hrsh7th/cmp-buffer'
Plug 'hrsh7th/cmp-nvim-lsp'
Plug 'hrsh7th/vim-vsnip'
Plug 'neovim/nvim-lspconfig'
call plug#end()
PlugInstall | quit

" Setup global configuration. More on configuration below.
lua << EOF
local cmp = require'cmp'

cmp.setup({
  snippet = {
    expand = function(args)
      vim.fn["vsnip#anonymous"](args.body)
    end,
  },
  view = {
    entries = {name = 'wildmenu'}
  },
  sources = cmp.config.sources({
    { name = 'nvim_lsp' },
    { name = 'vsnip' },
  }, {
    { name = 'buffer' },
  }),
})

cmp.setup({
  experimental = {
  ghost_text = true,
  },
})
EOF

lua << EOF
local capabilities = require('cmp_nvim_lsp').default_capabilities()

require'lspconfig'.pyright.setup {
  capabilities = capabilities,
}
EOF
```

Really, the only requirements in the above config are that the completion sources are set, and the wildmenu view is enabled.

Then, edit a new .py file and type `asset_`. As soon as you insert the underscore, the error will trigger. As of #1420 it will no longer trigger.
